### ISSUE TITLE
Triggering assignment rules on creation of a Lead?

### ISSUE THREAD

**Issue by coryrose1** [omniphx/forrest#203]
Hi,

I'm wanting to create Leads and trigger a Lead Assignment Rule to be ran.

In doing some research, it looks like a flag needs to be set in the header in order to achieve this: https://developer.salesforce.com/docs/atlas.en-us.api.meta/api/sforce_api_header_assignmentruleheader.htm?search_text=assignment

Is it possible to set this header in prior to inserting a lead with this package?

Thanks,
Cory

**Comment by coryrose1** [comment]
For reference, it looks like this is how it's done in the Salesforce PHP toolkit: https://developer.salesforce.com/index.php?title=PHP_Toolkit_11.0_AssignmentRuleHeader_Sample_(Partner)&redirect=no

**Comment by omniphx** [comment]
Hi @coryrose1, you can pass in headers in the same way you would set the method or body:
```
Forrest::sobjects('Account',[
    'method' => 'post',
    'body' => ['Name' => 'Dunder Mifflin'],
    'headers' => ['accept' => 'application/json']
]);
```

**Comment by coryrose1** [comment]
Thank you @omniphx ! Greatly appreciated.

**Comment by omniphx** [comment]
Apologies @coryrose1, jumped the gun on the solution without validating it first. Just tested out a scenario where I set up an assignment rule and realized I couldn't actually enable/disable it with a custom header. Luckily it was an easy fix and the above snippet should work in the latest v2.4.7 release.

I also updated the documentation to provide some examples for future reference. Thanks for bringing this up and providing a detailed description!
